File: exam/python/quick-sort.py
def partition(llist, low, high):  
    j=low
    pivot=llist[high]

    for i in range(low, high):
        if llist[i]<pivot:
            llist[i], llist[j] = llist[j], llist[i]
            j+=1
    (llist[j], llist[high])=(llist[high], llist[j])
    return j

def quick_sort(llist, low, high):
    if low<high:
        pivot_index=partition(llist, low, high)
        quick_sort(llist, low, pivot_index-1)
        quick_sort(llist, pivot_index+1, high)

    return llist

llist=[6,5,12,10,9,1]
print(quick_sort(llist, 0, len(llist)-1))

File: exam/algorithms/quick-sort/quick-sort.cpp
#include <iostream>
#include <vector>
int partition(std::vector<float>& arr, int low, int high){
    int j=low;
    float pivot=arr[high];
    for(int i =low; i<high; ++i){
        if(arr[i]<pivot){
            std::swap(arr[i], arr[j]);
            ++j;
        }
    }
    std::swap(arr[j], arr[high]);
    return j;
}

void QuickSort(std::vector<float>& arr, int low, int high){
    if(low<high){
        int pivot_index=partition(arr, low, high);
        QuickSort(arr, low, pivot_index-1);
        QuickSort(arr, pivot_index+1, high);
    }
}
int main(){
    int size;
    std::cout<<"Input size of array: "; std::cin>>size;
    std::vector<float> arr(size);

    for(int i=0; i<size; ++i){
        std::cout<<"arr["<<i<<"]=";
        std::cin>>arr[i];
    }

    QuickSort(arr, 0,size-1);
    
    for(int i=0; i<size; ++i){
        std::cout<<arr[i]<<" ";
    }
    std::cout<<std::endl;
    return 0;
}